src/js/hooks/cycleCount/useCountStep.jsx

```diff
@@ -302,15 +303,19 @@ const useCountStep = () => {
       show();
       resetValidationState();
       for (const cycleCount of tableData.current) {
-        const cycleCountItemsToUpdate = cycleCount.cycleCountItems.filter((item) => (item.updated && !item.id.includes('newRow')));
+        const cycleCountItemsToUpdate = cycleCount.cycleCountItems
+          .filter((item) => (item.updated && !item.id.includes('newRow')))
+          .map(trimLotNumberSpaces);
         for (const cycleCountItem of cycleCountItemsToUpdate) {
           await cycleCountApi.updateCycleCountItem(
             getPayload(cycleCountItem, cycleCount),
             currentLocation?.id,
             cycleCountItem?.id,
           );
         }
-        const cycleCountItemsToCreate = cycleCount.cycleCountItems.filter((item) => item.id.includes('newRow'));
+        const cycleCountItemsToCreate = cycleCount.cycleCountItems
+          .filter((item) => item.id.includes('newRow'))
+          .map(trimLotNumberSpaces);
```

Comment:
similar to my other comment, trimming whitespace here will prevent you from being able to use an existing lot  `"ABC  "` since it will always force the lot to be `"ABC"`.

I think this is fine though because to me it's a data issue that should be fixed. We shouldn't have lots in our system with trailing whitespace. That's confusing for users.

I think it's worth doing an investigation (in a different ticket) to see if there's an easy way to strip whitespace from all string fields coming into the frontend as well as all string fields going to the db. I can't see a situation where we'd need to allow trailing whitespace so maybe there's an easy way to remove it from our system entirely